File: backend/src/main/java/io/kongkham/kcurr/Controllers/FinancialNewsController.java
package io.kongkham.kcurr.Controllers;

import io.kongkham.kcurr.Models.FinancialNewsRequest;
import io.kongkham.kcurr.Models.FinancialNewsResponse;
import io.kongkham.kcurr.Services.FinancialNewsService;
import org.springframework.web.bind.annotation.*;

@RestController
public class FinancialNewsController {
    private final FinancialNewsService _financialNewsService;

    public FinancialNewsController(FinancialNewsService financialNewsService) {
        this._financialNewsService = financialNewsService;
    }

    @PostMapping("/news")
    public FinancialNewsResponse[] getFinancialNews(@RequestBody FinancialNewsRequest data) {
        String newsTopic = data.getNewsTopic();
        return _financialNewsService.getFinancialNews(newsTopic);
    }
}

File: backend/backend/Controllers/FinancialNewsController.cs
using backend.Models;
using backend.Services;
using Microsoft.AspNetCore.Mvc;

namespace backend.Controllers;

[ApiController]
public class FinancialNewsController : ControllerBase
{
    private readonly FinancialNewsService _financialNewsService;

    public FinancialNewsController(FinancialNewsService financialNewsService)
    {
        _financialNewsService = financialNewsService;
    }

    [HttpPost("news")]
    public async Task<FinancialNewsResponse[]> GetFinancialNews([FromBody] FinancialNewsRequest data)
    {
        string newsTopic = data.NewsTopic;
        return await _financialNewsService.GetFinancialNews(newsTopic);
    }
}